src/test/java/io/xstefank/wildfly/bot/PRAppendingMessage.java

```diff
@@ -0,0 +1,149 @@
+import static io.xstefank.wildfly.bot.helper.MockedGHPullRequestProcessor.mockCommits;
+import static io.xstefank.wildfly.bot.helper.MockedGHPullRequestProcessor.mockEmptyComments;
+import static io.xstefank.wildfly.bot.helper.MockedGHPullRequestProcessor.mockEmptyFileDetails;
+import static io.xstefank.wildfly.bot.helper.MockedGHPullRequestProcessor.processPullRequestMock;
+import static io.xstefank.wildfly.bot.util.Strings.blockQuoted;
+
+@QuarkusTest
+@GitHubAppTest
+public class PRAppendingMessage {
+
+    private String wildflyConfigFile = """
+            wildfly:
+              format:
+                commit:
+                  enabled: false
+            """;
+
+    private String appendedMessage = String.format("""
+            %s
+
+            %s%%s%s""", RuntimeConstants.BOT_MESSAGE_HEADER, blockQuoted(RuntimeConstants.BOT_JIRA_LINKS_HEADER), RuntimeConstants.BOT_MESSAGE_FOOTER);
+
+    @Test
+    public void testEmptyBodyAppendMessage() throws IOException {
+        GitHubJson githubJson = GitHubJson.builder("pr-template.json").action(Action.EDITED).title("WFLY-00000 title").body(null).build();
+
+        given().github(mocks -> {
+            mocks.configFile(RuntimeConstants.CONFIG_FILE_NAME).fromString(wildflyConfigFile);
+            processPullRequestMock(mocks.pullRequest(githubJson.pullRequest()), mockEmptyFileDetails(), mockEmptyComments(), mockCommits("WFLY-00000 commit"));
+        }).when().payloadFromString(githubJson.jsonString()).event(GHEvent.PULL_REQUEST).then().github(mocks -> {
+            StringBuilder sb = new StringBuilder();
+            Mockito.verify(mocks.pullRequest(githubJson.pullRequest())).setBody(String.format(appendedMessage,
+                    sb.append(blockQuoted(String.format(RuntimeConstants.BOT_JIRA_LINK_TEMPLATE, "WFLY-00000")).toString())));
+        });
+
+    }
+
+    @Test
+    public void testNonEmptyBodyAppendMessage() throws IOException {
+        String body = """
+                This is my
+                testing
+
+                body, which
+                should not be
+                cleared.""";
+        GitHubJson githubJson = GitHubJson.builder("pr-template.json").action(Action.EDITED).title("WFLY-00000 title").body(body).build();
+
+        given().github(mocks -> {
+            mocks.configFile(RuntimeConstants.CONFIG_FILE_NAME).fromString(wildflyConfigFile);
+            processPullRequestMock(mocks.pullRequest(githubJson.pullRequest()), mockEmptyFileDetails(), mockEmptyComments(), mockCommits("WFLY-00000 commit"));
+        }).when().payloadFromString(githubJson.jsonString()).event(GHEvent.PULL_REQUEST).then().github(mocks -> {
+            StringBuilder sb = new StringBuilder();
+            Mockito.verify(mocks.pullRequest(githubJson.pullRequest())).setBody(String.format(body + "\n\n" + appendedMessage,
+                    sb.append(blockQuoted(String.format(RuntimeConstants.BOT_JIRA_LINK_TEMPLATE, "WFLY-00000")).toString())));
```

Comment:
why do you need SB here?

Reply:
to be consistent with all the tests, rather then save on this one